### PR TITLE
Lazyload Comments

### PR DESCRIPTION
**Pull request by mattgreenham**
We're now lazyloading the comments module into the post page to avoid caching comments for the same ttl as the page

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: resources/js/blogs-bootstrap.js [hunk 1]
```diff
@@ -8,4 +8,14 @@ define(['istats-tracking', 'respimg', 'lazysizes'], function (IstatsTracking) {
     if ('querySelector' in document && 'addEventListener' in window) {
         init();
     }
+
+    $(function() {
```
Comment (tatitati) [inline resources/js/blogs-bootstrap.js]: The dollar sign is not necessary (I did the same mistake recently 😄 )

Comment (votemike) [inline resources/js/blogs-bootstrap.js]: Can this go in the init function?

File: src/Controller/CommentsIndexController.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        $post = $postService->getPostByGuid(new GUID($guid), $blog);
+
+        if (!$post) {
+            throw $this->createNotFoundException('Post not found');
+        }
+
+        $commentsPromise = $commentsService->getByBlogAndPost($blog, $post);
+
+        $this->hasVideo = $post->hasVideo();
```
Comment (votemike) [inline src/Controller/CommentsIndexController.php]: Where is this used? Is it used in the base controller somewhere?

File: templates/comments/index.html.twig [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}BBC {{ tr('page_title_blogs') }} - {{ blog.getName() }} - {{ post.getTitle() }} - {{ tr('link_comments') }}{% endblock %}
+
+{% block inline_head %}
+    {%- if comments -%}
```
Comment (votemike) [inline templates/comments/index.html.twig]: Would this comment page not have comments?

File: templates/comments/index.html.twig [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+{% block social_image_tags %}
+    {%- if blog.getImage() -%}
+        <meta property="og:image" content="{{ blog.getImage().getUrl(1200,675) }}" />
+    {%- else -%}
+        {{- parent() -}}
+    {%- endif -%}
+{% endblock %}
+
+{% block social_description %}
+    <meta property="og:description" content="{{ blog.getDescription() }}" />
```
Comment (votemike) [inline templates/comments/index.html.twig]: This isn't the blog itself is it? It's the comments for the blog? Is this description sensible?

File: templates/comments/partial.html.twig [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<div>
+    {%- if comments -%}
```
Comment (votemike) [inline templates/comments/partial.html.twig]: And if there aren't comments?

File: resources/js/blogs-bootstrap.js (hunk 1 above)
Comment (votemike) [reply resources/js/blogs-bootstrap.js]: Is that so? How come?

File: src/Controller/CommentsIndexController.php (hunk 2 above)
Comment (mattgreenham) [reply src/Controller/CommentsIndexController.php]: BaseController, for iStats labels

File: templates/comments/index.html.twig (hunk 3 above)
Comment (mattgreenham) [reply templates/comments/index.html.twig]: If the comments request promise fails then there wouldn't be

File: templates/comments/partial.html.twig (hunk 5 above)
Comment (mattgreenham) [reply templates/comments/partial.html.twig]: Have added a fallback

**Pull request merged**